### PR TITLE
Chapter 2 Homework 2: Inheritance

### PR DESCRIPTION
**Pull request by chatieb**
Chapter 2 Homework 2: Inheritance
This is a shop simulator, where player can buy and sell different items from their inventory. 
Player has inventory where they store their items. They can buy and sell items to/from a shop.
Shop sells/buys items. Shop can also just add and remove Item from a shop selection of merchandise.

### CODE REVIEW

File: Src/BootCamp.Chapter/Shop.cs [hunk 1]
```diff
@@ -12,10 +12,10 @@ public decimal GetMoney()
 
         public Shop()
         {
-
+            _inventory = new Inventory();
         }
 
-        public Shop(decimal money)
+        public Shop(decimal money) : this()
```
[the remark this answers is not in the document]
Comment (chatieb) [reply Src/BootCamp.Chapter/Shop.cs]: Changed code to handle bigger samples.  Thank you.

File: Src/BootCamp.Chapter/Inventory.cs [hunk 2]
```diff
@@ -15,12 +15,52 @@ public Inventory()
 
         public Item[] GetItems(string name)
         {
-            return new Item[0];
+            var numberOfMatchName = GetNumberOfMatchName(name);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Inventory.cs]: Good job for using this. Makes it efficient.

File: Src/BootCamp.Chapter/Inventory.cs [hunk 3]
```diff
@@ -15,12 +15,52 @@ public Inventory()
...
+                if (item.GetName() == name)
+                {
+                    tempItems[i++] = item;
+                }
+            }
+
+            return tempItems;
+        }
+
+        private int GetNumberOfMatchName(string name)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Inventory.cs]: Number is vague. Call "GetMatchingNamesCount"

File: Src/BootCamp.Chapter/Inventory.cs [hunk 4]
```diff
@@ -29,7 +69,47 @@ public void AddItem(Item item)
         /// </summary>
         public void RemoveItem(Item item)
         {
+            RemoveItem(item.GetName());
+        }
+
+        /// <summary>
+        /// Removes item matching criteria by item.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Inventory.cs]: Comment is less informative than method name. Can be removed.

File: Src/BootCamp.Chapter/Inventory.cs [hunk 5]
```diff
@@ -29,7 +69,47 @@ public void AddItem(Item item)
...
+                    {
+                        tempItems[i++] = item;
+                    }
+                }
+
+                _items = tempItems;
+            }
+        }
+
+        private int GetRemoveNumber(string name)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Inventory.cs]: Change Number to Count

Comment (chatieb) [reply Src/BootCamp.Chapter/Inventory.cs]: thxs for the suggestion

File: Src/BootCamp.Chapter/Inventory.cs (hunk 3 above)
Comment (chatieb) [reply Src/BootCamp.Chapter/Inventory.cs]: done

**Pull request closed without merging**